src/components/Structure3DModel/3DModel/index.tsx

```diff
@@ -76,15 +78,42 @@ const AlphaFoldModel = ({
 }: LoadedProps) => {
   const [shouldResetViewer, setShouldResetViewer] = useState(false);
   const [isReady, setReady] = useState(false);
+
+  // Added states for PDB availability check (moved from BFVDModelSubPage)
+  const [isPDBLoading, setIsPDBLoading] = useState(false);
+  const [isPDBAvailable, setIsPDBAvailable] = useState(false);
+  const [bfvdURL, setBfvdURL] = useState(bfvd || '');
+
+  // Effect to check PDB availability (moved from BFVDModelSubPage)
+  useEffect(() => {
+    setIsPDBLoading(true);
+    if (bfvd && proteinAcc.length > 0) {
+      setBfvdURL(bfvd);
+      fetch(bfvd, { method: 'HEAD' }).then((res) => {
+        if (res.status == 200) {
+          setIsPDBAvailable(true);
+        }
+      });
+    } else {
+      // If not using BFVD, assume AlphaFold models are available
+      setIsPDBAvailable(true);
+      setIsPDBLoading(false);
+    }
+    setIsPDBLoading(false);
```

Comment:
Call `setIsPDBLoading()` inside the callback of `fetch`, so after the promise has resolved, otherwise `setIsPDBLoading(false)` is called before the `fetch` returns anything, so we end up with `isPDBLoading` set to `false` and `isPDBAvailable` set to `true`, and the warning message (_Structure Viewer currently not available for this entry_) is temporary displayed.

```suggestion
        if (res.ok) {
          setIsPDBAvailable(true);
        }
        setIsPDBLoading(false);
      });
    } else {
      // If not using BFVD, assume AlphaFold models are available
      setIsPDBAvailable(true);
      setIsPDBLoading(false);
    }
```

Reply:
Yup, it was that way before the Loading fix commit and tried to fix in a different way. This suggestion doesn't work either because the `then` throws an error before executing  setIsPDBLoading(false). Handling it properly with a `catch` now.